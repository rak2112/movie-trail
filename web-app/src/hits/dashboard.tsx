import React from 'react';
import Movies, { MoviesProps } from '../all-movies/dashboard.container';

const Hits = (props: MoviesProps) =>  (<Movies {...props}/>);

export default Hits;
