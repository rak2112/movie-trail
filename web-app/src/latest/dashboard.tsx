import React from 'react';
import Movies, { MoviesProps } from '../all-movies/dashboard.container';

const Latest = (props: MoviesProps) =>  (<Movies {...props} />);

export default Latest;