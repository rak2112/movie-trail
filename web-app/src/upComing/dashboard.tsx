import React from 'react';
import Movies from '../all-movies/dashboard.container';

const UpComing = (props: any) =>  (<Movies {...props} route={'up-coming'}/>);

export default UpComing;
