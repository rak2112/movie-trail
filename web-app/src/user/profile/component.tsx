import React, { PureComponent } from 'react';
import { Nav, NavItem, Tab, Tabs } from 'react-bootstrap';
import styled from 'react-emotion';
import {
  MovieList,
  Toaster 
} from '../../core/components';
import { Profile } from './container';
import { MovieTabs } from './tabs';
import { ProfileDetails } from './user-details.component';


export class ProfileComponent extends PureComponent <Profile, any> {
  static displayName = 'MovieComponent';
  render() { 
    const { user } = this.props;
    return (
      <>
        <MovieList>
          {() => (
            <>
              <ProfileDetails {...user}/>
              <Toaster duration={2000}/>  
              <MovieTabs {...this.props}/>
            </>
          )}
        </MovieList>
      </>
    )
  }
}