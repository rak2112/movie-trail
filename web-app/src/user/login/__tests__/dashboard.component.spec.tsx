import { shallow } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter, Route } from 'react-router-dom';
import { create } from 'react-test-renderer';
import { combineReducers, createStore, Store } from 'redux'
import { reducer as form, reduxForm  } from 'redux-form'

import { FormProps, LoginComponent, LoginProps } from '../dashboard.component';

let store: any;
let props: any;

describe(`Login Component`, () => {
  beforeEach(() => {
    store = createStore(
      combineReducers({ form }),
      { form: {} }
    );

    props = {
      ...props,
      email: '',
      password: '',
      submitting: false,
      handleSubmit: jest.fn()
    };
  });

  it(`render the component, with form`, () => {

    const Decorated = reduxForm({ form: 'testForm' })(LoginComponent)
    const tree = create(
      <Provider store={store}>
        <MemoryRouter initialIndex={0} initialEntries={['/login']}>
          <Route path={'/login'} component={Decorated} />
        </MemoryRouter>
      </Provider>
    )
    expect(tree).toMatchSnapshot()
  });

  it(`render the component, should call onSubmit if form is submitted`, () => {

    const tree = shallow(
     <LoginComponent {...props}/>
    );

    tree.find('form').simulate('submit', {target: {email: ':email'}});
    expect(tree.find('form').props().onSubmit).toHaveBeenCalledWith({ target: { email: ':email' }});

  });
});