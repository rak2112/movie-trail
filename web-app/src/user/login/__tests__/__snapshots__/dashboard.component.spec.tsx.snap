// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component render the component, with form 1`] = `
<div
  className="css-1k2uugu"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <h3>
        Log In
      </h3>
      <div>
        <label />
        <div>
          <input
            className="form-control "
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            placeholder="User name"
            type="text"
            value=""
          />
          <em />
        </div>
      </div>
      <div>
        <label />
        <div>
          <input
            className="form-control "
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
          <em />
        </div>
      </div>
    </div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Log in
    </button>
    <a
      className="btn btn-warning"
      href="/movies"
      onClick={[Function]}
    >
      Cancel
    </a>
    <a
      className="nav-link"
      href="/register"
      onClick={[Function]}
    >
      Register
    </a>
    <a
      className="nav-link"
      href="/password-reset"
      onClick={[Function]}
    >
      Forgot Password?
    </a>
  </form>
</div>
`;
